sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -499,13 +499,21 @@ def __init__(self, evaluation_context, applied_ptransform,
     if self.source.id_label:
       raise NotImplementedError(
           'DirectRunner: id_label is not supported for PubSub reads')
+
+    import google.auth
+
+    _, sub_project = google.auth.default()
```

Comment:
Beam has this pipeline option, project which is generally used to specify project explicitly. Asking user explicitly for resource avoid surprises. Shall we use it. https://github.com/apache/beam/blob/40c2be03303d7c37cabfae4d4da6258a2ce33579/sdks/python/apache_beam/options/pipeline_options.py#L473

Reply:
Done. I'm defaulting to the default project from auth if the project is not specified. Thanks!